src/app/facility/utility-data/energy-consumption/utility-meter-data/edit-bill/edit-bill.component.ts

```diff
@@ -53,15 +65,38 @@ export class EditBillComponent implements OnInit {
           this.addOrEdit = 'edit';
           let accountMeterData: Array<IdbUtilityMeterData> = this.utilityMeterDataDbService.accountMeterData.getValue();
           this.editMeterData = accountMeterData.find(data => { return data.id == meterReadingId });
+          this.key = this.editMeterData.guid;
         } else {
           //new Reading
           let accountMeterData: Array<IdbUtilityMeterData> = this.utilityMeterDataDbService.accountMeterData.getValue();
           this.editMeterData = getNewIdbUtilityMeterData(this.editMeter, accountMeterData);
+          this.key = this.editMeterData.guid;
           this.addOrEdit = 'add';
         }
         this.setMeterDataForm();
       })
-    })
+    });
+
+    if (this.isElectron) {
```

Comment:
It looks like the same content is being added to each data-table component. 

Can you create a component that is just the button with the corresponding .ts logic in it. Then you can re-use the component in the different data tables?